pages/api/opprtunities/[id].ts

```diff
@@ -1,20 +1,64 @@
 import { prisma } from "@/lib";
+import { ApiResponseError, Router } from "@/lib/apiRouter";
+import { Curd } from "@/lib/crud";
 import { NextApiRequest, NextApiResponse } from "next";
 
+class SingleOpportunityController extends Curd<CrawledOpportunity | void>{
+    async Post(data: CrawledOpportunity):Promise<void> {
+        throw { message: 'This HTTP method is not allowed', statusCode: 400};
+    }
 
-export default async function handler(req:NextApiRequest, res:NextApiResponse) {
-    const {id}:Partial<{ [key: string]: string | string[]; }> = req.query
-    try{
-        if(typeof id === 'string'){
-            const job:CrawledOpportunity = await prisma.crawledOpportunity.findUniqueOrThrow({
-                where:{
-                    id: id
-                }
+    async Delete(data: CrawledOpportunity): Promise<CrawledOpportunity> {
+        try{
+            const deletedJob: CrawledOpportunity = await prisma.opportunity.delete({ where: { id: data.id } });
+            return deletedJob;
+        }
+        catch(error){
+            console.error(error);
+            throw error;
+        }
+    }
+
+    async Get(data: CrawledOpportunity): Promise<CrawledOpportunity> {
+        try {
+            const job: CrawledOpportunity = await prisma.opportunity.findFirstOrThrow({ where: { id: data.id } });
+            return job;
+        }
+        catch(error){
+            console.error(error);
+            if (error.code === 'P2025')
+                error.statusCode = 404;
+            throw error;
+        }
+    }
+
+    async Update(data: CrawledOpportunity): Promise<CrawledOpportunity> {
+        try{
+            const updatedJob: CrawledOpportunity = await prisma.opportunity.update({
+                data: data,
+                where: { id: data.id }
             });
-            return res.status(200).json(job)
+            return updatedJob;
+        }
+        catch(error){
+            console.error(error);
+            throw error;
         }
-  }
-  catch(err){
-    return res.status(500).json({message:"there's smth went wrong",Error:err})
-  }
+    }
+}
+
+export default async function handler(req: NextApiRequest, res: NextApiResponse) {
+    const controller = new SingleOpportunityController();
+    req.body ||= { id: req.query.id };
```

Comment:
this is not working

Reply:
Can you please explain in what case it is not working, as it is working locally fine with me

Reply:
the endpoint `opportunities/jsdkkldwqdkl123`

which is the id, and you can get it without using these, and you cannot be overrided, and this is a bad practice